### PR TITLE
Support blueprint static files per vhost

### PR DESCRIPTION
**Pull request by omerbn**

### CODE REVIEW

**Comment by omerbn** [comment]
Please help me understand how come it fails by Travis CI, but OK when i run pytest locally.

**Comment by r0fls** [comment]
I've faced this type of issue before. What I did was turn on gitlab ci on my own fork, and fiddle with things there.

File: sanic/app.py [hunk 1]
```diff
@@ -352,13 +352,13 @@ def middleware(self, middleware_or_request):
                            attach_to=middleware_or_request)
 
     # Static Files
-    def static(self, uri, file_or_directory, pattern=r'/?.+',
+    def static(self, uri, file_or_directory, host=None, pattern=r'/?.+',
```
Comment (seemethere) [inline sanic/app.py]: when adding a parameter it's always better to add at the end rather than in the middle

File: sanic/blueprints.py [hunk 2]
```diff
@@ -115,6 +114,7 @@ def decorator(handler):
                 name)
             self.routes.append(route)
             return handler
+
```
Comment (seemethere) [inline sanic/blueprints.py]: what's the point of the whitespace?

File: setup.py [hunk 3]
```diff
@@ -65,7 +66,7 @@ def open_local(paths, mode='r', encoding='utf8'):
     print("Installing without uJSON")
     requirements.remove(ujson)
 
-if strtobool(os.environ.get("SANIC_NO_UVLOOP", "no")):
+if platform == "win32" or strtobool(os.environ.get("SANIC_NO_UVLOOP", "no")):
```
Comment (seemethere) [inline setup.py]: This should be it's own PR

**Comment by r0fls** [comment]
Actually this should be closed per https://github.com/channelcat/sanic/pull/953. The original issue is here: https://github.com/channelcat/sanic/issues/949

@omerbn please reopen this, or the issue, if you see any problems.
